### ISSUE TITLE
Speedster optimize_model should raise error if input data is missing

### ISSUE THREAD

**Issue by Telemaco019** [nebuly-ai/nebullvm#239]
## Description

Invoking `optimize_model` with either `input_data=None` or `input_data=[]` generates the following output: 
```
2023-03-09 18:09:34 | INFO     | Running Speedster on CPU
``` 
However, the model is not optimized and no results are produced. Since `input_data` is necessary for optimizing the model, I think it would be useful to make it explicit by either raising an error or showing a proper log message when no input data is provided.

## How to replicate

Speedster version: 0.2.1
Nebullvm version: 0.8.1

```python
import torchvision.models as models
from speedster.api.functions import optimize_model


def main():
    optimize_model(
        model=models.resnet50(),
        input_data=[],
        store_latencies=True,
    )

if __name__ == '__main__':
    main()

```
